pkg/buildah/login.go

```diff
@@ -90,3 +135,29 @@ func loginCmd(c *cobra.Command, args []string, iopts *loginReply) error {
 	iopts.loginOpts.GetLoginSet = flagChanged(c, "get-login")
 	return auth.Login(ctx, systemContext, &iopts.loginOpts, args)
 }
+
+func GetCurrentUserFromKubeConfig(filename string) (userid string, err error) {
+	config, err := clientcmd.LoadFromFile(filename)
+	if err != nil {
+		return "", err
+	}
+	expectedCtx, exists := config.Contexts[config.CurrentContext]
+	if !exists {
+		return "", fmt.Errorf("failed to find current context %s", config.CurrentContext)
+	}
+	return expectedCtx.AuthInfo, nil
+}
+
+// If the specified string starts with http{s} it is replaced with it's
+// host[:port] parts; everything else is stripped. Otherwise, the string is
+// returned as is.
+func replaceURLByHostPort(repository string) (string, error) {
+	if !strings.HasPrefix(repository, "https://") && !strings.HasPrefix(repository, "http://") {
```

Comment:
Not as safe if passed by urls like 'ftp://sealos.io/../../root/.ssh/auth'.
 Maybe call `url.ParseRequestURI` and check return err.

Reply:
Any legitimate url could be a registry, and it's hard to use the filesystem to distinguish between different registries or different sealos-cloud-instance.
For example:
- https://hub.sealos.io/registry1
- https://hub.sealos.io/registry2/part2
or without scheme
- hub.sealos.io:5000
or
- yy.hub.sealos.io.cn


I was wondering if we should also maintain a jason file like docker, e.g.
```jason
{
	"registrys": {
		"hub.sealos.io:35000": {
			"kubeconfig": "...",
                        "user": "...",
		},
	},
        "clouds":{
        }
}
```
so, we can just edit and read it to get kubeconfig.


Or Based on userid uniqueness just maintain file like:$(HOME)/.sealos/users/$(userid)/config
The probability of different sealos generating the same userid and using it on the same cluster is very low.

Reply:
Use just `hostname` for path key is enough.

* http://hub.sealos.io/registry1 -> hub.sealos.io
* https://hub.sealos.io/registry2/part2 -> hub.sealos.io
* hub.sealos.io:5000 -> hub.sealos.io:5000
* yy.hub.sealos.io.cn -> yy.hub.sealos.io.cn

`http` and `https` refers to one domain.

We do not support one hotname with two or more registry, we can use:
* `registry1.hub.sealos.io`
* `registry2.hub.sealos.io` 

instead.